controllers/slicegateway/slicegateway.go

```diff
@@ -543,91 +573,94 @@ func (r *SliceGwReconciler) GetGwPodNameAndIP(ctx context.Context, sliceGw *kube
-		return "", ""
+		return []string{""}, []string{""}
 	}
-
+	podNameList := make([]string, 0)
+	podIPList := make([]string, 0)
 	for _, pod := range podList.Items {
 		if pod.Status.Phase == corev1.PodRunning {
-			return pod.Name, pod.Status.PodIP
+			podNameList = append(podNameList, pod.Name)
+			podIPList = append(podIPList, pod.Status.PodIP)
+			// return pod.Name, pod.Status.PodIP
 		}
 	}
 
-	return "", ""
+	return podNameList, podIPList
 }
 
 func isGatewayStatusChanged(slicegateway *kubeslicev1beta1.SliceGateway, podName string, podIP string, status *gwsidecar.GwStatus) bool {
-	return slicegateway.Status.PodName != podName ||
-		slicegateway.Status.PodIP != podIP ||
-		slicegateway.Status.LocalNsmIP != status.NsmStatus.LocalIP
+	return !contains(slicegateway.Status.PodNames, podName) ||
+		!contains(slicegateway.Status.PodIPs, podIP) ||
+		!contains(slicegateway.Status.LocalNsmIPs, status.NsmStatus.LocalIP)
 }
 
 func (r *SliceGwReconciler) ReconcileGwPodStatus(ctx context.Context, slicegateway *kubeslicev1beta1.SliceGateway) (ctrl.Result, error, bool) {
 	log := logger.FromContext(ctx).WithValues("type", "SliceGw")
 	debugLog := log.V(1)
 
-	podName, podIP := r.GetGwPodNameAndIP(ctx, slicegateway)
-	if podName == "" || podIP == "" {
-		log.Info("Gw pod not available yet, requeuing")
+	podNames, podIPs := r.GetGwPodNameAndIP(ctx, slicegateway)
+	if len(podNames) == 0 || len(podIPs) == 0 {
+		log.Info("Gw pods not available yet, requeuing")
 		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil, true
 	}
 
-	sidecarGrpcAddress := podIP + ":5000"
-
-	status, err := r.WorkerGWSidecarClient.GetStatus(ctx, sidecarGrpcAddress)
-	if err != nil {
-		log.Error(err, "Unable to fetch gw status")
-		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil, true
-	}
-
-	debugLog.Info("Got gw status", "result", status)
+	for i := 0; i < len(podNames); i++ {
+		sidecarGrpcAddress := podIPs[i] + ":5000"
 
-	if isGatewayStatusChanged(slicegateway, podName, podIP, status) {
-		log.Info("gw status changed")
-		slicegateway.Status.PodName = podName
-		slicegateway.Status.PodIP = podIP
-		slicegateway.Status.LocalNsmIP = status.NsmStatus.LocalIP
-		slicegateway.Status.ConnectionContextUpdatedOn = 0
-		err = r.Status().Update(ctx, slicegateway)
+		status, err := r.WorkerGWSidecarClient.GetStatus(ctx, sidecarGrpcAddress)
 		if err != nil {
-			log.Error(err, "Failed to update SliceGateway status for gateway status")
-			return ctrl.Result{}, err, true
+			log.Error(err, "Unable to fetch gw status")
+			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil, true
 		}
 
-		log.Info("gw status updated")
-	}
+		debugLog.Info("Got gw status", "result", status)
+
+		if isGatewayStatusChanged(slicegateway, podNames[i], podIPs[i], status) {
+			log.Info("gw status changed")
+			slicegateway.Status.PodNames = append(slicegateway.Status.PodNames, podNames[i])
+			slicegateway.Status.PodIPs = append(slicegateway.Status.PodIPs, podIPs[i])
```

Comment:
in case gw pod restarts , I think this will append new pod names and IPs to the list.
But it should update the new ones and remove the old ones

Reply:
Have made the change to this but its in a [different branch](https://github.com/kubeslice/worker-operator/tree/feature-ecmp-rebase)

Reply:
@Rahul-D78 i think its better if you raise a PR for feature-ecmp-rebase instead of feature-ecmp as it is already rebased with master